lib/scanner/blocks.go

```diff
@@ -105,9 +96,59 @@ func Blocks(ctx context.Context, r io.Reader, blocksize int, sizehint int64, cou
 		})
 	}
 
+	multiHashPool.Put(multiHf)
 	return blocks, nil
 }
 
+type multiHash struct {
+	strong hash.Hash
+	weak   hash.Hash32
+}
+
+var multiHashPool sync.Pool
+
+func newMultiHash(useWeakHashes bool) *multiHash {
+	h, ok := multiHashPool.Get().(*multiHash)
```

Comment:
Not sure what the purpose of this is, but we never put anything, so presumably it's always empty.
Also, we don't run many parallel Blocks() calls I think, the prevent read thrashing, so I wonder if there is any point in sync.Pool

Reply:
The Put is at the end of Blocks, just above the return statement.